docs/releasing.md

```diff
@@ -1,116 +1,141 @@
-<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+This document details the steps to create a release for
+`cluster-api-provider-metal3` aka CAPM3.
 
-- [Prerequisites](#prerequisites)
-   - [`docker`](#docker)
-- [Output](#output)
-  - [Expected artifacts](#expected-artifacts)
-  - [Artifact locations](#artifact-locations)
-- [Process](#process)
-  - [Permissions](#permissions)
+## Before making a release
 
-<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+Things you should do before making a release:
 
-## Prerequisites
+- Uplift controller to use latest corresponding CAPI modules
+- Uplift to latest corresponding IPAM module
+- Uplift to latest corresponding BMO module
+- Uplift any other direct/indirect dependency to close any public
+  vulnerabilities
 
-### `docker`
+## Permissions
 
-You must have docker installed.
+Creating a release requires repository `write` permissions for:
 
-## Output
+- Tag pushing
+- Branch creation
+- GitHub Release publishing
 
-### Expected artifacts
+These permissions are implicit for the org admins and repository admins.
+Release team gets their permissions via `metal3-release-team` membership. Adding
+person to the team gives him/her the necessary rights in all relevant
+repositories in the organization. Individual persons should not be added
+directly.
 
-1. A container image of the shared cluster-api-provider-metal3 manager
-1. A git tag
-1. A release on Github containing:
-     - A manifest file - `infrastructure-components.yaml`
-     - A metadata file - `metadata.yaml`
-     - A cluster template - `cluster-template.yaml`
-     - A file containing an example of variables to set - `example_variables.rc`
+## Process
 
-### Artifact locations
+CAPM3 uses [semantic versioning](https://semver.org). For version `v1.x.y`:
 
-1. The container image is found in the registry `quay.io/metal3-io` with an image
-   name of `cluster-api-provider-metal3` and a tag that matches the release
-   version. The image is automatically built once the release has been created.
+### Repository setup
 
-## Creating a release for CAPM3
+Clone out the repository:
+`git clone git@github.com:metal3-io/cluster-api-provider-metal`
 
-### Process
+or if using existing repository, verify which remote the `metal3-io`
+is: `git remote -v`. For this document, we assume it is `origin`.
 
-There are two ways to do the release: automated (using GitHub workflow) or manual.
-The processes are as follows:
+- If creating a new minor branch, identify the commit your wish to create the
+  branch from, and create a branch `release-1.x`:
+  `git checkout <sha> -b release-1.x` and push it to remote:
+  `git push origin release-1.x` to create it
+- If creating a new patch release, use existing branch `release-1.x`:
+  `git checkout origin/release-1.x`
 
-For version v0.x.y:
+### Tags
 
-#### Automated Process
+First we create a main tag, that triggers release note creation and
```

Comment:
main tag means ?

Reply:
Replaced with primary release tag, so its not confused with actual main branch.